src/constraints/outage_constraints.jl

```diff
@@ -143,14 +154,73 @@ function add_MG_fuel_burn_constraints(m,p)
+    @expression(m, ExpectedMGGenFuelCost,
+        sum( m[:dvMGGenMaxFuelCost][s] * p.s.electric_utility.outage_probabilities[s] for s in p.s.electric_utility.scenarios )
     )
+
+    m[:ExpectedMGFuelUsed] += ExpectedMGGenFuelUsed
+    m[:ExpectedMGFuelCost] += ExpectedMGGenFuelCost
 end
 
+function add_MG_CHP_fuel_burn_constraints(m, p; _n="")
+    # Fuel burn slope and intercept
+    fuel_burn_full_load = 1.0 / p.s.chp.electric_efficiency_full_load  # [kWt/kWe]
+    fuel_burn_half_load = 0.5 / p.s.chp.electric_efficiency_half_load  # [kWt/kWe]
+    fuel_burn_slope = (fuel_burn_full_load - fuel_burn_half_load) / (1.0 - 0.5)  # [kWt/kWe]
+    fuel_burn_intercept = fuel_burn_full_load - fuel_burn_slope * 1.0  # [kWt/kWe_rated]
+  
+    # Conditionally add dvFuelBurnYIntercept if coefficient p.FuelBurnYIntRate is greater than ~zero
+    if fuel_burn_intercept > 1.0E-7
+        #Constraint (1c1): Total Fuel burn for CHP **with** y-intercept fuel burn and supplementary firing
+        @constraint(m, MGCHPFuelBurnCon[t in p.techs.chp, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+            m[Symbol("dvMGFuelUsed"*_n)][t,s,tz]  == p.hours_per_time_step * 
+                m[Symbol("dvMGCHPFuelBurnYIntercept"*_n)][s,tz] +
+                sum(p.production_factor[t,tz+ts-1] * fuel_burn_slope * m[Symbol("dvMGRatedProduction"*_n)][t,s,tz,ts]
+                    for ts in 1:p.s.electric_utility.outage_durations[s])
+        )
+
+        #Constraint (1d): Y-intercept fuel burn for CHP across the scenario outage time steps
+        @constraint(m, MGCHPFuelBurnYIntCon[t in p.techs.chp, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+            m[Symbol("binMGCHPIsOnInTS"*_n)][s,tz,ts] => 
+                {m[Symbol("dvMGCHPFuelBurnYIntercept"*_n)][s,tz] >= sum(fuel_burn_intercept * m[Symbol("dvMGsize"*_n)][t] 
+                    for _ in 1:p.s.electric_utility.outage_durations[s])}
+        )
+    else
+        #Constraint (1c2): Total Fuel burn for CHP **without** y-intercept fuel burn
+        @constraint(m, MGCHPFuelBurnConLinear[t in p.techs.chp, s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+            m[Symbol("dvMGFuelUsed"*_n)][t,s,tz]  == p.hours_per_time_step *
+                sum(p.production_factor[t,tz+ts-1] * fuel_burn_slope * m[Symbol("dvMGRatedProduction"*_n)][t,s,tz,ts]
+                for ts in 1:p.s.electric_utility.outage_durations[s])
+        )
+    end 
+    
+    @constraint(m, [s in p.s.electric_utility.scenarios, tz in p.s.electric_utility.outage_start_time_steps],
+        m[:dvMGCHPMaxFuelUsage][s] >= sum( m[:dvMGFuelUsed][t, s, tz] for t in p.techs.chp )
+    )
+    
+    @expression(m, ExpectedMGCHPFuelUsed, 
+        sum( m[:dvMGCHPMaxFuelUsage][s] * p.s.electric_utility.outage_probabilities[s] for s in p.s.electric_utility.scenarios )
+    )
+
+    # fuel cost = kWh * $/kWh
```

Comment:
Seems weird that CHP dvMGFuelUsed units would be kwh but generator gal. Something to fix when we generalize generator to use other fuels. Made an issue.

Reply:
Yeah, I agree it's inconsistent but I'm also hesitant to keep adding issues for things that aren't necessarily issues right now. They are piling up (50+) and I'm concerned it's just getting to be a big pile of stuff we'll maybe never do.